30-indoor-air-quality-and-gas-leak/firmware/sparrow-application/aqi/aqi.cpp

```diff
@@ -0,0 +1,496 @@
+// Copyright 2023 Blues Inc.  All rights reserved.
+// Use of this source code is governed by licenses granted by the
+// copyright holder including that found in the LICENSE file.
+
+extern "C" {
+#include "aqi.h"
+
+// ST Header(s)
+#include <main.h>
+
+// Blues Header(s)
+#include <framework.h>
+#include <note.h>
+}
+
+#include "sfe_ens160.h"
+#include "sfe_bus.h"
+
+//
+// Configuration for the AQI app.
+//
+
+/**
+ * @brief Configures how often an air quality reading is taken and reported.
+ */
+#define AQI_MONITOR_PERIOD       60     // seconds
+
+/**
+ * @brief When set to true, only alerts are sent to notehub. Specifically, events with alert:1 or alert:3.
+ * When false, regular air quality events and alerts are sent.
+ */
+#define ALERTS_ONLY              (false)
+
+/**
+ * @brief Signal an alert when the AQI is equal or above this value.
+ * Set to 0 to disable alerts based on the AQI value.
+ */
+#define ALERT_AQI_LEVEL          3
+/**
+ * @brief  Signal an alert when the CO2 concentration (in ppm) is equal or above this value.
+ * Set to 0 to disable alerts based on the CO2 value.
+ */
+#define ALERT_ECO2_LEVEL         500    // ppm (atmospheric CO2 is ~420 ppm)
+
+/**
+ * @brief Signal an alert when the concentration of TVOCs (total volatile organic compounds)
+ * is at or above this level.
+ * Set to 0 to disable alerts based on the TVOC value.
+ */
+#define ALERT_TVOC_LEVEL         200    // ppb
+
+/**
+ * @brief How many times to try an I2C operation before considering it failed.
+ */
+#define AQI_I2C_RETRY_COUNT      3
+
+/**
+ * @brief How long to wait for an I2C response from the air quality sensor before 
+ * that attempt is considered unsuccessful.
+ * 
+ */
+#define AQI_I2C_TIMEOUT_MS       100
+
+
+// When set to `true` all notes are immediately synched to notehub. Useful for testing but not intended for production use.
```

Comment:
This one seems worth mentioning in the readme.

Reply:
I'm not sure what advice to give. 100ms seems pretty arbitrary.

Reply:
I mean mentioning that you can define `SYNC_ALL_NOTES` to true to get quicker feedback when testing. That seems worth mentioning in the readme, but it's not essential.